doc/user/spark.md

```diff
@@ -233,20 +242,73 @@ yarn.
 Then replicate `daos-site.xml`, `core-site.xml`, `yarn-site.xml` and
 `capacity-scheduler.xml` to other nodes.
 
-#### Access DAOS in Hadoop
+There is one pitfall that YARN runs with DAOS UNS URI, "daos:///uns-path",
+configured to `fs.defaultFS`. You may get an error complaining pool/container
+UUIDs cannot be found. It's because Hadoop considers the default filesystem is
+DAOS since you configured DAOS UNS URI. YARN has some working directories
+defaulting to local path without schema, like "/tmp/yarn", which is then
+constructed as "daos:///tmp/yarn". With this URI, Hadoop cannot connect to DAOS
+since no pool/container UUIDs can be found if daos-site.xml is not provided too
+.
 
-If everything goes well, you should see “/user” directory being listed after
-issuing below command.
+The simplest workaround for now is not set DAOS UNS URI to `fs.defaultFS`.
+
+#### Known Issues
+
+If you use Omni-path PSM2 provider in DAOS, you'll get connection issue in
+Yarn container due to PSM2 resource not being released properly in time.
+
+### Tune More Configurations
+
+If your DAOS URI is the non-UNS, you can follow descriptions of each
+config item to set your own values in loadable `daos-site.xml`.
+
+If your DAOS URI is the UNS path, your configurations, except those set by DAOS
+UNS creation, in `daos-site.xml` can still be effective. To make configuration
+source consistent, an alternative to the configuration file `daos-site.xml` is
+to set all configurations to the UNS path. You put the configs to the same UNS
+path with below command.
 
 ```bash
-$ hadoop fs -ls /
+# install attr package if get "command not found" error
+$ setfattr -n user.daos.hadoop -v "fs.daos.server.group=daos_server" <your path>
 ```
 
-You can also play around with other Hadoop commands, like -copyFromLocal and
--copyToLocal. You can also start Yarn and run some mapreduce jobs on Yarn. Just
-make sure you have DAOS URI, `daos://default:1/`, set correctly in your job.
+Or
 
-#### Known Issues
+```bash
+$ java -Dpath="your path" -Dattr=user.daos.hadoop -Dvalue="fs.daos.server.group=daos_server"
+    -cp ./daos-java-1.1.0-shaded.jar io.daos.dfs.DaosUns setappinfo
+```
 
-If you use Omni-path PSM2 provider in DAOS, you'll get connection issue in
-Yarn container due to PSM2 resource not being released properly in time.
+For the "value" property, you need to follow pattern, key1=value1:key2=value2..
+.. And key* should be from
+[daos-site-example.xml](hadoop-daos/src/main/resources/daos-site-example.xml).
+If value* contains characters of '=' or ':', you need to escape the value with
+below command.
+
+```bash
+$ java -Dop=escape-app-value -Dinput="daos_server:1=2" -cp ./daos-java-1.1.0-shaded.jar io.daos.dfs.DaosUns util
+```
+
+You'll get escaped value, "daos_server\u003a1\u003d2", for "daos_server:1=2".
+
+If you configure the same property in both `daos-site.xml` and UNS path, the
+value in `daos-sitem.xml` takes priority. If user set Hadoop configuration
```

Comment:
daos-sitem.xml -> daos-site.xml

Reply:
If user set -> If user sets or If you set

Reply:
changed.